lib/pure/net.nim

```diff
@@ -1312,9 +1318,14 @@ proc recv*(socket: Socket, data: var string, size: int, timeout = -1,
   if result < 0:
     data.setLen(0)
     let lastError = getSocketError(socket)
-    if flags.isDisconnectionError(lastError): return
+    if flags.isDisconnectionError(lastError):
+      when defineSsl:
+        if socket.isSsl:
+          socket.sslNoShutdown = true
+      return
```

Comment:
Would it make sense to simply put this in the `socketError` proc and avoid the duplication (+ handle other cases where an disconnection error occurs)?

Reply:
We can certainly put this in `socketError`, the problem would be that we need to change the API to accept `flags`, shouldn't be too hard I suppose.